filesystem/filesystem.go

```diff
@@ -3,62 +3,35 @@
 package filesystem
 
 import (
+	"context"
 	"fmt"
-	"log"
 	"os/exec"
 	"regexp"
 	"strings"
 	"time"
 )
 
 func getFileSystemInfo() (interface{}, error) {
-	/* Grab filesystem data from df	*/
-	cmd := exec.Command("df", dfOptions...)
-
-	outCh := make(chan []byte, 1)
-	errCh := make(chan error, 1)
 
-	var out interface{}
-	var err error
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
 
-	go func() {
-		_out, _err := cmd.Output()
-		if _err != nil {
-			errCh <- fmt.Errorf("df failed to collect filesystem data: %s", _err)
-			return
-		}
-		outCh <- _out
-	}()
+	/* Grab filesystem data from df	*/
+	cmd := exec.CommandContext(ctx, "df", dfOptions...)
 
-WAIT:
-	for {
-		select {
-		case res := <-outCh:
-			if res != nil {
-				out, err = parseDfOutput(string(res))
-			} else {
-				out, err = nil, fmt.Errorf("df failed to collect filesystem data")
-			}
-			break WAIT
-		case err = <-errCh:
-			out = nil
-			break WAIT
-		case <-time.After(2 * time.Second):
-			// Kill the process if it takes too long
-			if killErr := cmd.Process.Kill(); killErr != nil {
-				log.Fatal("failed to kill:", killErr)
-				// Force goroutine to exit
-				<-outCh
-			}
-		}
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("df failed to collect filesystem data: %s", err)
 	}
-
-	return out, err
+	if out != nil {
+		return parseDfOutput(string(out))
+	}
+	return nil, fmt.Errorf("df failed to collect filesystem data")
 }
 
 func parseDfOutput(out string) (interface{}, error) {
 	lines := strings.Split(out, "\n")
-	var fileSystemInfo []interface{} = make([]interface{}, len(lines)-2)
+	var fileSystemInfo = make([]interface{}, len(lines)-2)
```

Comment:
I don't know the behavior of `make([]interface{}, -2)` and probably won't happen, but I think it's worth adding a test on `len(lines)` like:

```
lines := strings.Split(out, "\n")
if len(lines) < 2 {
  return nil, fmt.Errorf("can't parse df output properly")
}
var fileSystemInfo = make([]interface{}, len(lines)-2)
```

Reply:
I've just checked the behavior and this:

```
str := ""
var a = make([]interface{}, len(str)-2)
```

panics.